src/validation_workflow/docker/validation_docker.py

```diff
@@ -30,15 +27,19 @@ def __init__(self, args: ValidationArgs) -> None:
 
     def download_artifacts(self) -> bool:
 
-        # Check if the docker daemon is running; or else, bring it up
         if (self.is_container_daemon_running()):
 
             # STEP 1 . pull the images for OS and OSD
-            self._OS_image_name = 'opensearchproject/opensearch' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch'
-            self._OSD_image_name = 'opensearchproject/opensearch-dashboards' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch-dashboards'
-
-            self.local_image_OS_id = self.get_image_id(self._OS_image_name, self.args.version)
-            self.local_image_OSD_id = self.get_image_id(self._OSD_image_name, self.args.version)
+            if (self.args.using_staging_artifact_only):
+                self._OS_image_name = 'opensearchstaging/opensearch' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchstaging/opensearch'
+                self._OSD_image_name = 'opensearchstaging/opensearch-dashboards' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchstaging/opensearch-dashboards'
+                self.local_image_OS_id = self.get_image_id(self._OS_image_name, ValidationArgs().stg_tag('opensearch').replace(" ", ""))
+                self.local_image_OSD_id = self.get_image_id(self._OSD_image_name, ValidationArgs().stg_tag('opensearch_dashboards').replace(" ", ""))
+            else:
+                self._OS_image_name = 'opensearchproject/opensearch' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch'
+                self._OSD_image_name = 'opensearchproject/opensearch-dashboards' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch-dashboards'
+                self.local_image_OS_id = self.get_image_id(self._OS_image_name, self.args.version)
+                self.local_image_OSD_id = self.get_image_id(self._OSD_image_name, self.args.version)
```

Comment:
Why do we add staging now?

Reply:
Thanks for the questions, @peterzhuamazon .. it was about 2 or 3 weeks ago, that we had a dry run, and there were recommendations to use only the staging artifact to run docker ( in the case we want to validate the release candidate prior of the formal release readiness ); also, the ability to run only the digest validation.